packages/devtools_app/lib/src/eval_on_dart_library.dart

```diff
@@ -95,28 +114,31 @@ class EvalOnDartLibrary {
 
   Future<InstanceRef> eval(
     String expression, {
-    @required ObjectGroup isAlive,
+    @required Disposable isAlive,
     Map<String, String> scope,
   }) {
     return addRequest(isAlive, () => _eval(expression, scope: scope));
   }
 
+  Future<LibraryRef> _waitForLibraryRef() async {
+    while (true) {
+      final libraryRef = await _libraryRef.future;
+      if (_libraryRef.isCompleted) {
```

Comment:
this is a good race condition to check for but you should check for it by checking whether the isolateRef matches the isolateRef when the method started.

Reply:
I didn't write this code. It was an "extract as method" from `eval`

Reply:
feel free to just make this a TODO for me to fix.